### DISCUSSION TITLE
Can't reset value of React Draft WYSIWYG using reset method of RHF

### DISCUSSION THREAD

**Discussion by ngurahyudi** [react-hook-form/react-hook-form#7814] [Q&A] [answered]
Hi Bill @bluebill1049 
I'm trying to integrate React Draft WYSIWYG with RHF using controller, everything works well until i try to reset the  WYSIWYG after successfully submitting the form. The form value of WYSIWYG is resets, but the text won't

Here is my codesandbox
https://codesandbox.io/s/damp-platform-pss4r?file=/src/App.js:1758-2007

Thank You

**Comment by bluebill1049** [comment, marked as answer]
here is a working example: https://codesandbox.io/s/white-cookies-qw7v8?file=/src/App.js

you will have to get the content from the submitted data `EditorState`, I am not familiar with the library. I think the important aspect here is to wire up the `value` and `onChange`, when you work with react hook form, you normally don't need an external `state`.

```tsx
import "./styles.css";
import { useForm, Controller } from "react-hook-form";
import { Editor } from "react-draft-wysiwyg";
import "react-draft-wysiwyg/dist/react-draft-wysiwyg.css";

export default function App() {
  const defaultValues = {
    someText: "",
    someDraft: null
  };

  const { handleSubmit, control, reset } = useForm({
    defaultValues
  });

  const onSubmit = (data) => {
    console.log(data);
  };

  const onError = (data) => console.log("err: ", data);

  return (
    <div className="App">
      <form onSubmit={handleSubmit(onSubmit, onError)}>
        <Controller
          name="someText"
          control={control}
          render={({ field }) => {
            return (
              <input
                {...field}
                onChange={(e) => {
                  const params = e.target.value.toUpperCase();
                  field.onChange(params);
                }}
              />
            );
          }}
        />

        <Controller
          name="someDraft"
          control={control}
          render={({ field }) => {
            return (
              <Editor
                editorStyle={{
                  padding: "0px 10px 10px",
                  height: "200px"
                }}
                editorState={field.value} // wire up the value
                wrapperClassName="wrapper-class"
                editorClassName="editor-class"
                onEditorStateChange={field.onChange} // send data with the onChagne
              />
            );
          }}
        />

        <button type="submit">submit</button>
        <button type="buton" onClick={() => reset()}>
          reset
        </button>
      </form>
    </div>
  );
}
````

**Reply by ngurahyudi** [reply]
Thanks for your reply Bill, I'm trying to figure it out!

**Reply by ngurahyudi** [reply]
Hi @bluebill1049 I think I found the solution
for anyone who might have similar problem, please refer into this codesandbox link:

[sandbox](https://codesandbox.io/s/zen-shirley-xws62?file=/src/App.js)

Thanks for the clue, Bill :)

Regards

**Reply by bluebill1049** [reply]
well done 👍  would you like to leave your answer in another comment and mark it as the answer to help others?

**Reply by ngurahyudi** [reply]
> well done 👍 would you like to leave your answer in another comment and mark it as the answer to help others?

will do, Sir..
